docs/plugins/advanced_topics/widget_communication.md

```diff
@@ -0,0 +1,146 @@
+(widget-communication)=
+
+# Accessing docked widgets
+
+Sometimes more complex workflows require access to other docked widgets, sometimes from other plugins, created by other developers.
+
+## Accessing plugin widgets `viewer.window.add_plugin_dock_widget`
+
+If a plugin widget is already docked in the viewer, 
+calling the `add_plugin_dock_widget` method will return the existing widget instance.
+
+This is the most convenient way to access a plugin widget that is required by your plugin.
+Because if the widget is absent, it will be created and added to the viewer.
+
+## Accessing widget by name `viewer.window.get_dock_widget`
+
+If you need to access a widget, but without creating it, you can use the `get_dock_widget` method.
+The `get_dock_widget` method allows you to retrieve a docked widget by its name.
+
+This method returns the `QtViewerDockWidget` associated with the widget, or `None` if the widget is not found.
+
+To access the original widget, you can use the `inner_widget` method of the `QtViewerDockWidget`.
+
+This method allows accessing non-plugin widgets added to viewer using the `add_dock_widget` method.
+
+
+*The `get_dock_widget` and `inner_widget` were added in napari 0.6.2.*
+
+
+## Widget name 
+
+When widget is added to the viewer as plugin contribution (by using a menu or `add_plugin_dock_widget`), it is assigned a name.
+The name is created by concatenating the widget name and the plugin name in brackets, like this: `"widget_name (plugin_name)"`.
```

Comment:
```suggestion
The name is created by concatenating the widget name and the plugin name in brackets, like this: `"Widget Name (plugin_name)"`.
```
I was confused by the underscore, thinking it could be the Python name.